File: ToolLib/NotCaptureLib/src/main/java/com/yc/notcapturelib/proxy/ProxyWifiUtils.java
package com.yc.notcapturelib.proxy;

import android.content.Context;
import android.os.Build;
import android.text.TextUtils;

/**
 *    @author yangchong
 *    GitHub : https://github.com/yangchong211/YCAppTool
 *    time   : 2020/11/30
 *    desc   : 代理工具类
 */
public final class ProxyWifiUtils {


    /**
     * 判断设备 是否使用代理上网
     * @param context                               上下文
     * @return                                      设备是否链接代理
     */
    public static boolean isWifiProxy(Context context) {
        // 是否大于等于4.0
        final boolean IS_ICS_OR_LATER = Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH;
        String proxyAddress;
        int proxyPort;
        if (IS_ICS_OR_LATER) {
            proxyAddress = System.getProperty("http.proxyHost");
            String portStr = System.getProperty("http.proxyPort");
            proxyPort = Integer.parseInt((portStr != null ? portStr : "-1"));
        } else {
            proxyAddress = android.net.Proxy.getHost(context);
            proxyPort = android.net.Proxy.getPort(context);
        }
        return (!TextUtils.isEmpty(proxyAddress)) && (proxyPort != -1);
    }


}

File: ToolLib/NotCaptureLib/src/main/java/com/yc/notcapturelib/ProxyWifiUtils.kt
package com.yc.notcapturelib

import android.content.Context
import android.os.Build
import android.text.TextUtils
import android.util.Log

class ProxyWifiUtils {

    fun checkWifiProxy(context: Context): Boolean {
        val IS_ICS_OR_LATER = Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH
        val proxyAddress: String?
        val proxyPort: Int?
        if (IS_ICS_OR_LATER) {
            proxyAddress = System.getProperty("http.proxyHost")
            val portStr = System.getProperty("http.proxyPort")
            proxyPort = Integer.parseInt(portStr ?: "-1")
        } else {
            proxyAddress = android.net.Proxy.getHost(context)
            proxyPort = android.net.Proxy.getPort(context)
        }
        Log.i("ProxyWifi","proxyAddress : ${proxyAddress}, prot : ${proxyPort}")
        return !TextUtils.isEmpty(proxyAddress) && proxyPort != -1
    }


}